beacon_node/network/src/sync/manager.rs

```diff
@@ -675,6 +670,8 @@ impl<T: BeaconChainTypes> SyncManager<T> {
                 if new_state.is_synced() {
                     self.backfill_sync.fully_synced_peer_joined();
                 }
+
+                self.network.update_peer_sync_status(peer_id, new_state);
```

Comment:
Previously we would update the status even if it was of the same kind, whereas now we only update in the case where they are different. Couldn't this cause issues with the sync status of peers becoming stale? E.g. If a peer sends us a status update from `SyncStatus::Synced { sync_info1 }` to `SyncStatus::Synced { sync_info2 }` don't we want to update the internal info?

Apologies if this is way off-track, I'm just looking for any substantial change in this PR that could have caused the increase in simulator failures since this was merged (see: https://github.com/sigp/lighthouse/commits/unstable). @divagant-martian's comment [here](https://github.com/sigp/lighthouse/pull/2724#issuecomment-974017031) also isn't exactly reassuring :sweat_smile:

Reply:
Ah, thinking this might be what Diva was referring to [here](https://github.com/sigp/lighthouse/pull/2724#issuecomment-948037692)?

> sync depending on the info being always synced.